### ISSUE TITLE
No controller input

### ISSUE THREAD

**Issue by kaysur** [Plagman/gamescope#438]
I set up gamescope on Fedora and when launching a game with it I have no controller input, however when using the `-e` flag I have controller input.
gamescope wiki implies this is for Steam integration, so is this somehow related to controller input working? I don't have Steam Controller Input enabled for my games, so this doesn't make sense to me.
I think this flag should be properly documented in the wiki and README with other flags because I already gave up on troubleshooting gamescope on one day and knowing this would have helped.

**Comment by aqxa1** [comment]
I have the same issue, but the workaround isn't needed for all games, so it might just be an incompatibility with the particular game. E.g. it's broken with `Chrono Trigger` but not `Desperados III` or `Ion Fury`, for example. Possibly a Proton issue given both of the working games are native releases.

**Comment by kaysur** [comment]
https://github.com/Plagman/gamescope/issues/420
Looks like this could be related. First report says mouse works in Proton experimental, but not 7.0-1, so when I play another round of Hades or something I can also try that.

**Comment by AtilaSaraiva** [comment]
It doesn't even work here with the -e option. I'm using the flatpak steam version with NixOS as the OS.

**Comment by Holzhaus** [comment]
I had the same issue with Rise of the Tomb Raider. When running it via Proton 7.0-1, mouse and controller input works, but when using gamescope, both mouse and controller stop working (keyboard input still works though).

I confirm that adding the `-e` flag fixes the problem:

    gamescope -W 1920 -H 1080 -e -- %command%

**Comment by Holzhaus** [comment]
Just a note, I have Steam Input enabled (because I'm mainly playing via Steam Link).

@cactushydrocodone maybe check if you have Steam Input enabled for PS4 controllers?

**Comment by cactushydrocodone** [comment]
> Just a note, I have Steam Input enabled (because I'm mainly playing via Steam Link).
> 
> @cactushydrocodone maybe check if you have Steam Input enabled for PS4 controllers?

Damn that helped! Now i can press any button, which brings me to the controller configuration thingy of the game which looks a bit funky:
![image](https://user-images.githubusercontent.com/12892508/162053000-996a65b4-09e6-48a1-b99c-59f25beda6ab.png)

Now, i cant press anything except the touchpad button anymore. :/

**Comment by cactushydrocodone** [comment]
Its not only LEGO Star Wars, none of the games i tested work.

**Comment by cactushydrocodone** [comment]
I FINALLY solved it.
It wasnt a gamescope problem, it was a steam problem!
I turned off the controller support for playstation and suddenly, it worked!

Those are my settings in steam now:
![image](https://user-images.githubusercontent.com/12892508/164909495-e6785c52-ea58-4bff-a890-84e4cf4abaca.png)

**Comment by opale95** [comment]
I have the problem on games ran with Proton at any higher version than 6.3-8.
Keyboard and Mouse seems to work, but not my Xbox360 wired controller.
Anything Steam Input related is disabled on my end (Globally and per-game).
I tried disabling Steam in-game overlay or launching games via Big Picture Mode with no luck.
I checked it was working fine without `gamescope`.

**Comment by cactushydrocodone** [comment]
@opale95 -e is not documented but it is required for every proton version after 6.3. Just use it, it does not hurt ;)

**Comment by QushyQushy** [comment]
I have the same issue with the Neptune Controller on the Steam Deck. In HZD neither the DualSense or Neptune controllers work. In Cybeprpunk the DualSense controller works (even though no Playsation Controller compatibility is enabled in Big picture Mode) but the Neptune doesn't. This is the command I'm using:

`gamescope -e -U -f -w 960 -h 600 -W 1280 -H 800 -- mangohud %command%`

**Comment by Hydranet** [comment]
I'm having the same issue with my mouse when using gamescope with Proton 7.x, so I was currently running al my games on Proton 6.3.8 where I wasn't have the issue. The solution mentioned here is to launch gamescope with the -e option, does anyone know what changed  in Proton or gamescope that this option is needed?

@cactushydrocodone How did you find out that -e is needed for every Proton version after 6.3.x?

**Comment by cactushydrocodone** [comment]
> I'm having the same issue with my mouse when using gamescope with Proton 7.x, so I was currently running al my games on Proton 6.3.8 where I wasn't have the issue. The solution mentioned here is to launch gamescope with the -e option, does anyone know what changed in Proton or gamescope that this option is needed?
> 
> @cactushydrocodone How did you find out that -e is needed for every Proton version after 6.3.x?

In every issue similar to this people say "use -e" and it works. Thats how i found out :P

**Comment by Bitwolfies** [comment]
> does anyone know what changed in Proton or gamescope that this option is needed?
Honestly I wish I knew, but I hope this is simply a temporary bandaid and it will actually be resolved, after all, nothings more permanent then a temporary fix.

**Comment by sidebuster** [comment]
I played State of Decay 2 an hour ago then came back to it an suddenly I couldn't use my mouse or Steam controller. Then I tried an old 360 controller and it still didn't work. 

Tried -e and nothing changed.
Disabled "Playstation Configuration Support" in the general controller settings and it worked again.

**Comment by Tarch1** [comment]
Use -e option with Elden Ring make controller work again (and have steam overlay working).

**Comment by shaumux** [comment]
I still get the issue if I enable steam input, if I disable steam input the controller works.

I'm using Gentoo and Proton-tkg along with a xbox one controller.

**Comment by ddulic** [comment]
I have an issue where I can achieve smooth performance only when running `-b`, but that is causing the controller to no longer work, even with the `-e` flag enabled as well.

More info in https://github.com/ValveSoftware/steam-for-linux/issues/7130#issuecomment-1217767064

**Comment by UltraBlackLinux** [comment]
I have a similar problem. Controller support does not work. Steam integration just causes the gamescope window to never render.
Regardless, I am using https://github.com/kozec/sc-controller to handle my controller inputs, and not the native steam handler

**Comment by AndyDurden** [comment]
I'm on the steam deck, in desktop mode, running games from normal desktop steam with launch options:
`gamescope -- %command%`
(some games won't launch without -e, some games won't launch with -e)

My steaminput controller configuration is never applied in this case. When I run games without any launch options, steaminput controller configurations are applied, and when I tab in/out it will tab to the respective game/desktop controller configuration. I assume it should work the same way when explicitly running through gamescope? Tested on hl2, disgaea 2, and dungeon siege. I was originally trying to do this to get controller configurations to work on non-steam applications launched through steam, but as far as I can tell there's no way to do that atm.

**Comment by minionprocyk** [comment]
Chiming in here. I have a dell xps 15 9500 with a gtx 1650 ti and an Intel CometLake-H GT2. The 2 gpus are relevant because my nvidia gpu cannot load gamescope properly and fails during initialization. I have to load gamescope using the intel gpu Some games input worked, some games don't. The following were important fixes for me to get a working experience.

-  Use -e option to get input working in some games but only with steam -gamepadui (regular steam without the option will load, but fail to display any window at all and the terminal will spam 'bad window parameter'. Furthermore, you cannot use gamescope from launch parameters within the steam game, it only works if I use gamescope for steam itself, and only if the -gamepadui option is enabled, no other configuration will work)
- export INTEL_DEBUG=norbc otherwise my intel gpu will display a mosaic hue and nothing will look correct at all
- FSR, NIS both work but tax the integrated gpu too much so typically keep those disabled unfortunately :(. What will happen is the game will render at great framerates, all inputs will be handled gracefully, but the display/dekstop itself will lag, it's a weird experience.  

TLDR I wrapped up a shell script that looks like this and now I can play black mesa and alt-tab or un-fullscreen without crashing. I hope my nvidia gpu works with this in the future so i can try rendering 1920x1200 and upscale to 3840x2400.
 
```

#!/bin/bash
export INTEL_DEBUG=norbc
export mesa_glthread=true
exec gamescope -W 1920 -H 1200 -w 1440 -h 90 -r 60 -o 15 -U -e -- steam -gamepadui

```

**Comment by Max-Pare** [comment]
when using the -e flag my games open but there is no display, I can hear them running in the background but they don't display

**Comment by awsms** [comment]
Now facing this bug, my Xbox One controller isn't recognized when running a game through gamescope.
Using xpadneo-dkms + Hyprland.

**Comment by paxscull** [comment]
On Nobara and encountering this issue. If I set the `-e` the window for the fix, the windows is invisible when it starts.  

My Xbox Controller  isn't passing normal gamepad controls but interestingly enough its following the "Non-Game Control Layouts" if the gamepad is mapped to Keyboard controls .

**Comment by ThatOneBonk** [comment]
> On Nobara and encountering this issue. If I set the `-e` the window for the fix, the windows is invisible when it starts.
> 
> My Xbox Controller isn't passing normal gamepad controls but interestingly enough its following the "Non-Game Control Layouts" if the gamepad is mapped to Keyboard controls .

Same issue with a PS4 controller. Steam Input breaks when using gamescope, nothing seems to fix it, `-e` prevents the game window from rendering. This is both for Proton and native games. Extremely unfortunate, and I hope that someone will be able to take a look at it. Gentoo Linux.
